maas/client/flesh/__init__.py

```diff
@@ -511,8 +512,18 @@ def post_mortem(traceback):
     post_mortem(traceback)
 
 
+def program_name_from_env(program):
+    """Return the program name from environment."""
+    if os.environ.get("SNAP_INSTANCE_NAME"):
+        return os.environ.get("SNAP_INSTANCE_NAME")
+    if os.environ.get("SNAP_NAME"):
+        return os.environ.get("SNAP_NAME")
```

Comment:
you shouldn't need this, `SNAP_INSTANCE_NAME` is always there for the snap

Reply:
Okay will remove it, didn't know if that would always be the case, if you had a older snapd.

Reply:
I think that got introduced way before content interface :)